packages/unmock-core/src/generator-experimental.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * Implements the logic for generating a response from a service file
+ */
+// Try fixing broken imports in Node <= 8 by using require instead of default import
+const jsf = require("json-schema-faker"); // tslint:disable-line:no-var-requires
+import { array } from "fp-ts/lib/Array";
+import { isNone, none, Option, some } from "fp-ts/lib/Option";
+import { isParameter, isResponse, MediaType } from "loas3/dist/generated/full";
+import { fromTraversable, Getter, Iso, Lens, Optional, Prism } from "monocle-ts";
+import {
+  allMethods,
+  changeRef,
+  changeRefs,
+  getComponentFromRef,
+  getParameterFromRef,
+  getResponseFromRef,
+  getSchemaFromRef,
+  internalGetComponent,
+  MethodNames,
+  objectToArray,
+  valueLens,
+} from "openapi-refinements";
+import * as url from "url";
+import {
+  CreateResponse,
+  IListener,
+  ISerializedRequest,
+  ISerializedResponse,
+  IUnmockOptions,
+} from "./interfaces";
+import {
+  Header,
+  isReference,
+  OpenAPIObject,
+  Operation,
+  Parameter,
+  PathItem,
+  Reference,
+  Response,
+  Responses,
+  Schema,
+} from "./service/interfaces";
+import { ServiceStore } from "./service/serviceStore";
+
+export const without = <T>(p: T, m: keyof T): T => {
+  const o = { ...p };
+  delete o[m];
+  return o;
+};
+
+export const matchUrls = (protocol: string, host: string, o: OpenAPIObject): string[] =>
+  o.servers
+    ? o.servers.map(m => m.url).filter(i => new url.URL(i).host === host && new url.URL(i).protocol === `${protocol}:`)
+    : [];
+
+const prunePathItemInternal = (m: MethodNames, a: MethodNames[], p: PathItem): PathItem =>
+  a.length === 0 ? p : prunePathItemInternal(m, a.slice(1), a[0] === m ? p : without(p, a[0]));
+
+export const prunePathItem = (m: MethodNames, p: PathItem) => prunePathItemInternal(m, allMethods, p);
+
+export const matchesInternal = (a: string[], b: string[]): boolean =>
+  a.length === b.length
+  && (a.length === 0
+    || (((a[0] === b[0])
+      || (b[0].length > 2 && b[0][0] === "{" && b[0].slice(-1) === "}")) && matchesInternal(a.slice(1), b.slice(1))));
```

Comment:
These internal methods could maybe use some documentation and/or examples, it's quite hard to follow what they're doing 🤔